src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -730,21 +730,19 @@ def add_fluorescence_traces(
         plane_index=plane_index,
     )
 
-    rate = (
-        np.nan
-        if segmentation_extractor.get_sampling_frequency() is None
-        else float(segmentation_extractor.get_sampling_frequency())
-    )
-
     roi_response_series_kwargs = dict(
         rois=roi_table_region,
-        # TODO: timestamps or rate (for timestamps we need frame_to_time method)
-        # TODO: check for regularity, only use timestamps if irregular timing
-        starting_time=0.0,
-        rate=rate,
         unit="n.a.",
     )
 
+    # Add timestamps or rate
+    timestamps = segmentation_extractor.frame_to_time(np.arange(segmentation_extractor.get_num_frames()))
+    rate = calculate_regular_series_rate(series=timestamps)
+    if rate:
+        roi_response_series_kwargs.update(starting_time=timestamps[0], rate=rate)
+    else:
+        roi_response_series_kwargs.update(timestamps=H5DataIO(data=timestamps, **compression_options))
```

Comment:
@CodyCBakerPhD looked into the tests once more and doing

        self.assertAlmostEqual(
            df_over_f[trace_name].rate,
            segmentation_extractor.get_sampling_frequency(),
            places=6
        )

doesn't fix the tests, because  `df_over_f[trace_name].rate` is `None`.

Reply:
Are the timestamps for the dummy extractor in that test irregular? I didn't think they were, but if the `rate` isn't being written to the object in the tests that's the only reason I can think as to why... does the object in the failing tests have `timestamps` instead?

Reply:
yeah they have `timestamps` instead in this case, but still I thought it is a rounding issue because if we use `tolerance_decimals = 5` when trying to infer rate from timestamps in `calculate_regular_series_rate`, then it's regular.

Reply:
OK thanks that helps pin down the issue.

I think the actual problem is with the `SegmentationExtractor.frame_to_time()` method here: https://github.com/catalystneuro/roiextractors/blob/master/src/roiextractors/segmentationextractor.py#L305

which if you compare to SpikeInterface (which doesn't have this problem) does not round the timestamps: https://github.com/catalystneuro/spikeinterface/blob/master/spikeinterface/core/baserecording.py#L448-L449